src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java

```diff
@@ -148,4 +156,65 @@ public String getCmekKey(Arguments arguments) {
     }
     return arguments.get("gcp.cmek.key.name");
   }
+
+  public CryptoKeyName parseCmekKey(String key, FailureCollector collector) {
```

Comment:
Both methods have different use cases. `parseCmekKey()` is used to validate cmek key formatted string while `getCmekKey()` is used to fetch the cmek key formatted string. I will move the `parseCmekKey()` method to CmekUtils class. This will resolve the conflict.

Reply:
Is there is no reason to get the raw string anymore? Seems like the caller should always get a valid object (from which you can get the resource uri if needed)